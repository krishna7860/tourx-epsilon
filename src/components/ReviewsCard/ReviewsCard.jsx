/* eslint-disable react/prop-types */
import { Tooltip } from "@material-ui/core";
import React from "react";
import StarRating from "../StartRating/StarRating";
import {
  ReviewCardWrap,
  Wrapper,
  TopSection,
  Date,
  RatingWrap,
  Rating,
  UserDetail,
  ImageWarp,
  Deatils,
  Img,
  ReviewsSection,
  RatingWrapper,
  Readmore,
} from "./Style";

function ReviewsCard(props) {
  const { review } = props;
  return (
    <ReviewCardWrap>
      <Wrapper>
        <TopSection>
          <Date>{review?.createdAt?.split("T")[0]}</Date>
          <RatingWrap>
            <Rating>Excellent</Rating>
            <RatingWrapper>
              <StarRating
                rating={review?.rating ?? 5}
                color="#C0980A"
                width="20px"
                height="20px"
              />
            </RatingWrapper>
          </RatingWrap>
        </TopSection>
        <UserDetail>
          <ImageWarp>
            <Img src={review?.imgUrl} alt="imgProfile" />
          </ImageWarp>
        </UserDetail>
        <div>
          <Deatils>{review?.username}</Deatils>
          <ReviewsSection>{`${review?.description.slice(
            0,
            200
          )}... `}</ReviewsSection>
          {review?.description?.length > 200 ? (
            <Tooltip
              title={review?.description}
              aria-label="description"
              style={{ fontSize: "14px" }}
            >
              <Readmore>Read more</Readmore>
            </Tooltip>
          ) : null}
        </div>
      </Wrapper>
    </ReviewCardWrap>
  );
}

export default ReviewsCard;
