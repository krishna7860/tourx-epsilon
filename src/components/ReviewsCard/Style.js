import styled from "styled-components";

export const ReviewCardWrap = styled.div`
  position: relative;
  padding-top: 33.5px;
  margin: 20px;
`;
export const Wrapper = styled.div`
  background-color: #fff;
  padding: 16px 16px 23.6px;
  box-shadow: 0px 1px 4px rgb(0 0 0 / 25%);
  border-radius: 10px;
  height: 240px;
`;

export const TopSection = styled.div`
  justify-content: space-between;
  margin-bottom: 12px;
  display: flex;
`;
export const Date = styled.div`
  color: #848484;
  @media screen and (max-width: 1024px) {
    font-size: 14px;
  }
  @media screen and (max-width: 768px) {
    font-size: 12px;
  }
`;

export const RatingWrap = styled.div`
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
`;
export const RatingWrapper = styled.div`
  margin-right: 5px;
`;
export const Rating = styled.div`
  margin-bottom: 6px;
  color: #000;
`;

export const UserDetail = styled.div`
  position: absolute;
  top: 0;
  left: calc(50% - 45px);
`;

export const ImageWarp = styled.div`
  height: 74px;
  width: 74px;
  background-color: #fff;
  border-radius: 50%;
  margin-bottom: 20px;
`;
export const Img = styled.img`
  position: absolute;
  top: 2px;
  left: calc(50% - 37px);
  height: 70px;
  width: 70px;
  border-radius: 50%;
  display: inline-block;
  -o-object-fit: cover;
  object-fit: cover;
`;
export const Deatils = styled.div`
  font-size: 16px;
  font-weight: bold;
  color: #4d4c4c;
  text-align: center;
  margin-bottom: 8px;
`;
export const ReviewsSection = styled.div`
  color: #868686;
  line-height: 1.54;
  text-align: center;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
  margin-bottom: 20px;
  /* max-height: 64.68px; */
  transition: 0.2s max-height;
  overflow: hidden;
`;

export const Readmore = styled.p`
  text-align: center;
  font-size: 14px;
`;
