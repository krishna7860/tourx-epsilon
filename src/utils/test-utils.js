/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from "react";
import { render as rtlRender } from "@testing-library/react";
// import { renderHook as rtlRenderHooks } from "@testing-library/react-hooks";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import * as themes from "../theme/schema.json";
import reducer from "../redux/rootReducer";
// import defaultStore from "../redux/defaultStore";

function render(ui, options = {}) {
  const {
    initialState = {},
    store = createStore(reducer, initialState),
    theme,
    ...renderOptions
  } = options;

  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <ThemeProvider theme={themes.data.light}>{children}</ThemeProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// function renderHookWithRedux(callback, options = {}) {
//   const { initialState = defaultStore, store = createStore(reducer, initialState), ...others } = options

//   const returned = rtlRenderHooks(callback, {
//     ...others,
//     // eslint-disable-next-line react/display-name
//     wrapper: ({ children }) => <Provider store={store}>{children}</Provider>
//   })

//   return {
//     ...returned,
//     store
//   }
// }

// re-export everything
export * from "@testing-library/react";
// override render method
export { render, rtlRender as renderHook };
