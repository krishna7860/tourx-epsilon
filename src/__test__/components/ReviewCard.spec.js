/* eslint-disable react/jsx-filename-extension */
import React from "react";
import { render } from "../../utils/test-utils";
import ReviewsCard from "../../components/ReviewsCard/ReviewsCard";

describe("components/ReviewsCard", () => {
  it("should be rendered properly", () => {
    const { asFragment } = render(<ReviewsCard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
