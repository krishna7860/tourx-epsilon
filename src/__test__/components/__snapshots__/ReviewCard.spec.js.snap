// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ReviewsCard should be rendered properly 1`] = `
<DocumentFragment>
  <div
    class="sc-dlnjwi iMXIIi"
  >
    <div
      class="sc-hKFxyN hYeCAv"
    >
      <div
        class="sc-eCApnc kqbBPp"
      >
        <div
          class="sc-jSFjdj iMOYkM"
        >
          25 May 2021
        </div>
        <div
          class="sc-gKAaRy liRQYl"
        >
          <div
            class="sc-fujyAs kmrHkY"
          >
            Excellent
          </div>
          <div
            class="sc-iCoGMd cZgeQ"
          >
            <div
              class="sc-gtsrHT eUTdid"
            >
              <span
                class="sc-bdnxRM kqFglc"
                color="#C0980A"
                height="20px"
                rotate="0"
                width="20px"
              >
                <svg
                  class=""
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z"
                  />
                </svg>
              </span>
              <span
                class="sc-bdnxRM kqFglc"
                color="#C0980A"
                height="20px"
                rotate="0"
                width="20px"
              >
                <svg
                  class=""
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z"
                  />
                </svg>
              </span>
              <span
                class="sc-bdnxRM kqFglc"
                color="#C0980A"
                height="20px"
                rotate="0"
                width="20px"
              >
                <svg
                  class=""
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M394 480a16 16 0 01-9.39-3L256 383.76 127.39 477a16 16 0 01-24.55-18.08L153 310.35 23 221.2a16 16 0 019-29.2h160.38l48.4-148.95a16 16 0 0130.44 0l48.4 149H480a16 16 0 019.05 29.2L359 310.35l50.13 148.53A16 16 0 01394 480z"
                  />
                </svg>
              </span>
              <span
                class="sc-bdnxRM kqFglc"
                color="#C0980A"
                height="20px"
                rotate="0"
                width="20px"
              >
                <svg
                  class=""
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                    fill="none"
                    stroke="currentColor"
                    stroke-linejoin="round"
                    stroke-width="32"
                  />
                  <path
                    d="M256 48v316L118 464l54-160-140-96h172l52-160z"
                  />
                </svg>
              </span>
              <span
                class="sc-bdnxRM kqFglc"
                color="#C0980A"
                height="20px"
                rotate="0"
                width="20px"
              >
                <svg
                  class=""
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M480 208H308L256 48l-52 160H32l140 96-54 160 138-100 138 100-54-160z"
                    fill="none"
                    stroke="currentColor"
                    stroke-linejoin="round"
                    stroke-width="32"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-pNWdM huaUaz"
      >
        <div
          class="sc-jrsJWt jJUkxK"
        >
          <img
            alt="imgProfile"
            class="sc-kEqXSa bEPfyl"
            src="https://res.cloudinary.com/thrillophilia/image/upload/c_fill,f_auto,fl_progressive.strip_profile,g_auto,h_300,q_auto,w_400/v1/filestore/euqkvteayahctk0onr7842oo3adx_profile_image.jpg"
          />
        </div>
        <div
          class="sc-iqAclL cLbgZD"
        >
          Marut Butt
        </div>
      </div>
      <div>
        <div
          class="sc-crzoAE GcTWJ"
        >
          Highly recommended! I booked for my stays at the Serai Resort and I can proudly say that my stays were worth the money. Thrillophilia helped
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
